astropy/units/core.py

```diff
@@ -1328,14 +1327,38 @@ def cgs(self):
     @property
     def physical_type(self):
         """
-        Return the physical type on the unit.
+        Return the physical type(s) that are dimensionally compatible
```

Comment:
Maybe keep as one line as "Physical type(s) dimensionally compatible with the unit."

Reply:
@mhvk — Thank you!  Your comments have been really helpful in improving this pull request on both the small scale code changes as well as the larger overall structure.  I'll hopefully have a chance to implement the suggestions by early this week.

> Also, maybe time to squash the commits?

That does seem like an increasingly good idea.  We could either squash and merge when the PR is ready, or I could do a rebase and squash commits individually.